File: Exemplos/App_eXperience_XamarinAndroid/Elgin_AppExperience_XamarinAndroid-font/Impressora/FragmentPrinterBarCode.cs
using Android.Support.V4.App;
using Android.OS;
using Android.Views;
using Android.Widget;
using System.Collections.Generic;

namespace M8
{
    public class FragmentPrinterBarCode : Fragment
    {
        EditText editCodigo;
        Spinner spinnerTipoCodigo, spinnerHeight, spinnerWidth;
        RadioButton radioEsquerda, radioCentralizado, radioDireita;
        CheckBox checkboxCutPaper;
        Button btnImprimirCodbar;

        Dictionary<string, string> parametros;

        TextView textviewHeigth;

        public override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            // Create your fragment here
        }

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
        {
            base.OnCreateView(inflater, container, savedInstanceState);
            // Use this to return your custom view for this Fragment
            View fragmentView = inflater.Inflate(Resource.Layout.fragmentprinterbarcode, container, false);

            //EditTexts
            editCodigo = fragmentView.FindViewById<EditText>(Resource.Id.editCodigo);

            //Spinners
            spinnerTipoCodigo = fragmentView.FindViewById<Spinner>(Resource.Id.spinnerTipoCodigo);
            spinnerHeight = fragmentView.FindViewById<Spinner>(Resource.Id.spinnerHeight);
            spinnerWidth = fragmentView.FindViewById<Spinner>(Resource.Id.spinnerWidth);

            //Radios
            radioEsquerda = fragmentView.FindViewById<RadioButton>(Resource.Id.radioEsquerdaBarcode);
            radioCentralizado = fragmentView.FindViewById<RadioButton>(Resource.Id.radioCentralizadoBarcode);
            radioDireita = fragmentView.FindViewById<RadioButton>(Resource.Id.radioDireitaBarcode);

            //Checkboxes
            checkboxCutPaper = fragmentView.FindViewById<CheckBox>(Resource.Id.checkboxCutPaperBarcode);

            //Buttons
            btnImprimirCodbar = fragmentView.FindViewById<Button>(Resource.Id.btnImprimirBarcode);

            textviewHeigth = fragmentView.FindViewById<TextView>(Resource.Id.textViewHeight);

            InitialState();

            InitViewsFunc();


            return fragmentView;
        }

        private void InitialState()
        {
            editCodigo.Text = "40170725";
            radioCentralizado.Checked = true;
            spinnerTipoCodigo.SetSelection(0);
            spinnerHeight.SetSelection(2);
            spinnerWidth.SetSelection(5);
            
        }

        private void InitViewsFunc()
        {
            spinnerTipoCodigo.ItemSelected += delegate
            {
                spinnerHeight.Visibility = ViewStates.Visible;
                textviewHeigth.Visibility = ViewStates.Visible;
                if (spinnerTipoCodigo.SelectedItem.ToString() == "EAN 8") editCodigo.Text = "40170725";
                else if (spinnerTipoCodigo.SelectedItem.ToString() == "EAN 13") editCodigo.Text = "0123456789012";
                else if (spinnerTipoCodigo.SelectedItem.ToString() == "QR Code")
                {
                    editCodigo.Text = "ELGIN DEVELOPERS COMMUNITY";
                    spinnerHeight.Visibility = ViewStates.Invisible;
                    textviewHeigth.Visibility = ViewStates.Invisible;
                }
                else if (spinnerTipoCodigo.SelectedItem.ToString() == "UPC-A") editCodigo.Text = "123601057072";
                else if (spinnerTipoCodigo.SelectedItem.ToString() == "UPC-E") editCodigo.Text = "1234567";
                else if (spinnerTipoCodigo.SelectedItem.ToString() == "CODE 39") editCodigo.Text = "*ABC123*";
                else if (spinnerTipoCodigo.SelectedItem.ToString() == "ITF") editCodigo.Text = "05012345678900";
                else if (spinnerTipoCodigo.SelectedItem.ToString() == "CODE BAR") editCodigo.Text = "A3419500A";
                else if (spinnerTipoCodigo.SelectedItem.ToString() == "CODE 93") editCodigo.Text = "ABC123456789";
                else editCodigo.Text = "{C1233";
            };

            btnImprimirCodbar.Click += delegate
            {
                parametros = new Dictionary<string, string>();
                if (spinnerTipoCodigo.SelectedItem.ToString() == "QR Code")
                {
                  
                    parametros.Add("qrSize", spinnerWidth.SelectedItem.ToString());
                    parametros.Add("text", editCodigo.Text);

                    if (radioEsquerda.Checked == true) parametros.Add("align", "Esquerda");
                    else if (radioCentralizado.Checked == true) parametros.Add("align", "Centralizado");

                    else parametros.Add("align", "Direita");

                    Printer.ImprimeQR_CODE(parametros);
                    Printer.AvancaLinhas(10);
                    if (checkboxCutPaper.Checked) Printer.CutPaper(10);
                }
                else
                {
                    parametros.Add("barCodeType", spinnerTipoCodigo.SelectedItem.ToString());
                    parametros.Add("text", editCodigo.Text);
                    parametros.Add("height", spinnerHeight.SelectedItem.ToString());
                    parametros.Add("width", spinnerWidth.SelectedItem.ToString());

                    if (radioEsquerda.Checked == true) parametros.Add("align", "Esquerda");
                    else if (radioCentralizado.Checked == true) parametros.Add("align", "Centralizado");
                    else parametros.Add("align", "Direita");

                    Printer.ImprimeBarCode(parametros);
                    Printer.AvancaLinhas(10);
                    if (checkboxCutPaper.Checked) Printer.CutPaper(10);
                }

            };


        }
    }
}

File: Exemplos/App_eXperience_Java/Elgin_AppExperience_Java-font/app/src/main/java/com/elgin/elginexperience/FragmentPrinterBarCode.java
package com.elgin.elginexperience;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.app.Fragment;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Spinner;
import android.widget.TextView;

import java.util.HashMap;
import java.util.Map;

public class FragmentPrinterBarCode extends Fragment {
    //DECLARE EDIT TEXT
    EditText editTextInputBarCode;

    //DECLARE SPINNER
    Spinner spinnerBarCodeType;
    Spinner spinnerBarCodeWidth;
    Spinner spinnerBarCodeHeight;

    //DECLARE RADIO BUTTONS
    RadioGroup radioGroupAlignBarCode;
    RadioButton buttonRadioAlignCenter;

    //DECLARE CHECKBOX
    CheckBox checkBoxIsCutPaperBarCode;

    //DECLARE TEXTVIEW
    TextView textViewWidth, textViewHeight;

    //DECLARE BUTTONS
    Button buttonPrinterBarCode;

    //DECLARE VALUES CONFIGS
    String typeOfBarCode = "EAN 8";
    String typAlignOfBarCode = "Centralizado";
    int widthOfBarCode = 1;
    int heightOfBarCode = 20;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View v = inflater.inflate(R.layout.activity_fragment_printer_bar_code, container, false);

        //INIT EDIT TEXT
        editTextInputBarCode = v.findViewById(R.id.editTextInputBarCode);
        editTextInputBarCode.setText("40170725");

        textViewWidth = v.findViewById(R.id.textViewWidth);
        textViewHeight = v.findViewById(R.id.textViewHeight);


        //INIT RADIOS, SPINNER AND BUTTONS
        spinnerBarCodeType = v.findViewById(R.id.spinnerBarCodeType);
        buttonRadioAlignCenter = v.findViewById(R.id.radioButtonBarCodeAlignCenter);
        spinnerBarCodeWidth = v.findViewById(R.id.spinnerBarCodeWidth);
        spinnerBarCodeHeight = v.findViewById(R.id.spinnerBarCodeHeight);
        checkBoxIsCutPaperBarCode = v.findViewById(R.id.checkBoxCutPaper);
        buttonPrinterBarCode = v.findViewById(R.id.buttonPrintBarCode);

        //CONFIGS FONT FAMILY
        spinnerBarCodeType.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onNothingSelected(AdapterView<?> parent) { }

            @Override
            public void onItemSelected(AdapterView adapter, View v, int i, long lng) {
                typeOfBarCode = adapter.getItemAtPosition(i).toString();
                setTypeCodeMessage(typeOfBarCode);
            }
        });

        //CONFIGS ALIGN BAR CODE
        buttonRadioAlignCenter.setChecked(true);
        radioGroupAlignBarCode = v.findViewById(R.id.radioGroupAlignBarCode);
        radioGroupAlignBarCode.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @SuppressLint("NonConstantResourceId")
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                if (checkedId == R.id.radioButtonBarCodeAlignLeft) {
                    typAlignOfBarCode = "Esquerda";
                    System.out.println(typAlignOfBarCode);
                } else if (checkedId == R.id.radioButtonBarCodeAlignCenter) {
                    typAlignOfBarCode = "Centralizado";
                    System.out.println(typAlignOfBarCode);
                } else if (checkedId == R.id.radioButtonBarCodeAlignRight) {
                    typAlignOfBarCode = "Direita";
                    System.out.println(typAlignOfBarCode);
                }
            }
        });



        //CONFIGS WIDTH BAR CODE
        spinnerBarCodeWidth.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onNothingSelected(AdapterView<?> parent) { }

            @Override
            public void onItemSelected(AdapterView adapter, View v, int i, long lng) {
                widthOfBarCode = Integer.parseInt(adapter.getItemAtPosition(i).toString());
            }
        });

        //CONFIGS HEIGHT BAR CODE
        spinnerBarCodeHeight.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onNothingSelected(AdapterView<?> parent) { }

            @Override
            public void onItemSelected(AdapterView adapter, View v, int i, long lng) {
                heightOfBarCode = Integer.parseInt(adapter.getItemAtPosition(i).toString());
            }
        });

        //BUTTON PRINT BAR CODE CLICKED
        buttonPrinterBarCode.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(typeOfBarCode.equals("QR CODE")){
                    sendPrinterQrCode();
                }else{
                    sendPrinterBarCode();
                }
            }
        });

        return v;
    }

    public void setTypeCodeMessage(String typeActual){
        textViewWidth.setText("WIDTH");
        textViewHeight.setVisibility(View.VISIBLE);
        spinnerBarCodeHeight.setVisibility(View.VISIBLE);

        switch(typeActual){
            case "EAN 8":
                editTextInputBarCode.setText("40170725");
                break;
            case "EAN 13":
                editTextInputBarCode.setText("0123456789012");
                break;
            case "QR CODE":
                textViewWidth.setText("SQUARE");
                textViewHeight.setVisibility(View.INVISIBLE);
                spinnerBarCodeHeight.setVisibility(View.INVISIBLE);

                editTextInputBarCode.setText("ELGIN DEVELOPERS COMMUNITY");
                break;
            case "UPC-A":
                editTextInputBarCode.setText("123601057072");
                break;
            case "CODE 39":
                editTextInputBarCode.setText("CODE39");
                break;
            case "ITF":
                editTextInputBarCode.setText("05012345678900");
                break;
            case "CODE BAR":
                editTextInputBarCode.setText("A3419500A");
                break;
            case "CODE 93":
                editTextInputBarCode.setText("CODE93");
                break;
            case "CODE 128":
                editTextInputBarCode.setText("{C1233");
                break;
        }
    }

    public void sendPrinterBarCode(){
        Map<String, Object> mapValues = new HashMap<>();

        if(editTextInputBarCode.getText().toString().equals("")) {
            alertMessage();
        }else{
            mapValues.put("barCodeType", typeOfBarCode);
            mapValues.put("text", editTextInputBarCode.getText().toString());
            mapValues.put("height", heightOfBarCode);
            mapValues.put("width", widthOfBarCode);
            mapValues.put("align", typAlignOfBarCode);

            int result = PrinterMenu.printer.imprimeBarCode(mapValues);
            System.out.println("RESULT BAR CODE: " + result);
            PrinterMenu.jumpLine();
            if (checkBoxIsCutPaperBarCode.isChecked()) PrinterMenu.cutPaper();
        }
    }

    public void sendPrinterQrCode(){
        Map<String, Object> mapValues = new HashMap<>();

        if(editTextInputBarCode.getText().toString().equals("")) {
            alertMessage();
        }else{
            mapValues.put("qrSize", widthOfBarCode);
            mapValues.put("text", editTextInputBarCode.getText().toString());
            mapValues.put("align", typAlignOfBarCode);

            PrinterMenu.printer.imprimeQR_CODE(mapValues);
            PrinterMenu.jumpLine();
            if (checkBoxIsCutPaperBarCode.isChecked()) PrinterMenu.cutPaper();
        }
    }


    public void alertMessage(){
        AlertDialog alertDialog = new AlertDialog.Builder(PrinterMenu.printer.mActivity).create();
        alertDialog.setTitle("Alert");
        alertDialog.setMessage("Campo código vazio.");
        alertDialog.setButton(AlertDialog.BUTTON_NEUTRAL, "OK",
                new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.dismiss();
                    }
                });
        alertDialog.show();
    }
}